src/Flow/ETL/Transformer/EntryValueTransformer.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Transformer;
+
+use Flow\ETL\Exception\RuntimeException;
+use Flow\ETL\Row;
+use Flow\ETL\Row\EntryFactory;
+use Flow\ETL\Row\Factory\NativeEntryFactory;
+use Flow\ETL\Rows;
+use Flow\ETL\Transformer;
+
+/**
+ * @psalm-immutable
+ */
+final class EntryValueTransformer implements Transformer
+{
+    private static ?bool $isSerializable = null;
+
+    /**
+     * @var array<string>
+     */
+    private array $entries;
+
+    private string $userFunction;
+
+    private EntryFactory $entryFactory;
+
+    /**
+     * @param array<string> $entries
+     * @param string $userFunction
+     */
+    public function __construct(array $entries, string $userFunction, EntryFactory $entryFactory = null)
```

Comment:
If you change string type and rename `$userFunction` into `callable` it should be possible to provide it as a:

- \Closure - `fn($value) : \do_something($value)`
- string - `strtoupper` 
- array - `[$this, 'somePublicMethod']` or `[SomeClass::class, 'someStaticPublicMethod']`

```
    /**
     * @param array<string> $entries
     * @param callable $callback
     */
    public function __construct(array $entries, callable $callback, EntryFactory $entryFactory = null)
    {
        $this->entries = $entries;
        $this->callback = $callback;
        $this->entryFactory = $entryFactory ?? new NativeEntryFactory();
    }
```

This should make transformer a lot more flexible, but during serialization you would need to check if $callback is \Closure and then use opis/closure library if available, and if not, throw an exception. Same with deserialization.